flask_security/core.py

```diff
@@ -494,37 +512,26 @@ def init_app(self, app, datastore=None, register_blueprint=True,
 
         identity_loaded.connect_via(app)(_on_identity_loaded)
 
-        state = _get_state(app, self.datastore,
-                           login_form=login_form,
-                           confirm_register_form=confirm_register_form,
-                           register_form=register_form,
-                           forgot_password_form=forgot_password_form,
-                           reset_password_form=reset_password_form,
-                           change_password_form=change_password_form,
-                           send_confirmation_form=send_confirmation_form,
-                           passwordless_login_form=passwordless_login_form,
-                           anonymous_user=anonymous_user)
-
-        if register_blueprint:
-            app.register_blueprint(create_blueprint(state, __name__))
+        self._state = _get_state(app, self.datastore, **self.kwargs)
+
+        if self.kwargs['register_blueprint']:
+            app.register_blueprint(create_blueprint(self._state, __name__))
             app.context_processor(_context_processor)
 
         @app.before_first_request
         def _register_i18n():
             if '_' not in app.jinja_env.globals:
-                app.jinja_env.globals['_'] = state.i18n_domain.gettext
```

Comment:
keep the changes minimal by using trick in https://github.com/mattupstate/flask-security/pull/703/files#r143207783